components/TextFields/src/ContainedInputView/MDCBaseTextField.m

```diff
@@ -265,15 +315,76 @@ - (CGRect)rightViewRectForBounds:(CGRect)bounds {
 }
 
 - (CGRect)textRectForBounds:(CGRect)bounds {
-  return [self adjustTextAreaFrame:self.layout.textRect
+  CGRect textRect = [self textRectFromLayout:self.layout labelState:self.labelState];
+  return [self adjustTextAreaFrame:textRect
       withParentClassTextAreaFrame:[super textRectForBounds:bounds]];
 }
 
 - (CGRect)editingRectForBounds:(CGRect)bounds {
-  return [self adjustTextAreaFrame:self.layout.textRect
+  CGRect textRect = [self textRectFromLayout:self.layout labelState:self.labelState];
+  return [self adjustTextAreaFrame:textRect
       withParentClassTextAreaFrame:[super textRectForBounds:bounds]];
 }
 
+- (UIFont *)normalFont {
+  return self.font ?: [self uiTextFieldDefaultFont];
+}
+
+- (UIFont *)floatingFont {
+  return [self.normalFont fontWithSize:(self.normalFont.pointSize * (CGFloat)0.5)];
+}
+
+- (UIFont *)uiTextFieldDefaultFont {
+  static dispatch_once_t onceToken;
+  static UIFont *font;
+  dispatch_once(&onceToken, ^{
+    font = [UIFont systemFontOfSize:[UIFont systemFontSize]];
+  });
+  return font;
+}
+
+#pragma mark Label
+
+- (BOOL)canLabelFloat {
```

Comment:
shouldLabelFloat ?

Reply:
It's really like, "can it _ever_ float", not "should it float right now"